src/scripts.js

```diff
@@ -2,7 +2,205 @@ import './styles.css';
+viewAllButton.classList.add("hidden");
+recipeRepo.data.forEach((recipe) => {
+  allRecipeList.innerHTML += `<li class="recipe-in-list" id=${recipe.id}>${recipe.name}</li>`
+})
+}
+
+const searchBtn = document.querySelector('.search-btn')
+
+searchBtn.addEventListener('click', searchRecipe)
+
+function searchRecipe() {
+    filteredByTagArea.innerHTML = ''
+    let searchInput = document.querySelector('.searchbar').value
+    searchInput = searchInput.toLowerCase();
+    recipeRepo.data.forEach((recipe) => {
+        if (recipe.name.toLowerCase().includes(searchInput)) {
+            createFilteredArea();
+        filteredByTagArea.innerHTML += `
+        <h1 id=${recipe.id}>${recipe.name}</h1>
+        `
+        }
+    })
+    document.querySelector('.searchbar').value = ''
+}
+
+function displayRecipe(event) {
+  createRecipeArea();
+  populateRecipeArea();
+}
+
+function createRecipeArea() {
+  foodImagesSection.classList.add("hidden");
+  allRecipesSection.classList.add("hidden");
+  favoriteRecipeArea.classList.add("hidden")
+  viewAllButton.classList.add("hidden");
+  displayRecipeSection.classList.remove("hidden")
+  displayRecipeSection.innerHTML = ''
+  filteredByTagArea.classList.add("hidden")
+}
+
+function createFilteredArea() {
+  foodImagesSection.classList.add("hidden");
+  allRecipesSection.classList.add("hidden");
+  favoriteRecipeArea.classList.add("hidden")
+  viewAllButton.classList.add("hidden");
+  displayRecipeSection.classList.add("hidden")
+  displayRecipeSection.innerHTML = ''
+  filteredByTagArea.classList.remove("hidden")
+}
+
+function populateRecipeArea() {
+  recipeRepo.data.forEach((recipe) => {
+    if(event.target.id === recipe.id.toString()) {
+      let recipeToDisplay = new Recipe(recipe)
+      let recipeCost = recipeToDisplay.getRecipeCost()
+      recipeToDisplay.createIngredientList()
+      const officialIngredients = recipeToDisplay.ingredientsList.map((ingredient) => {
+        return ` ${ingredient.name}`
+      })
+      let recipeInstructions = recipe["instructions"]
+      let officialInstructions = recipeInstructions.map((instruction) => {
+        return instruction.instruction
+      }).flat()
+      displayRecipeSection.innerHTML += `
+      <img src="${recipeToDisplay.image}">
+      <h1>${recipeToDisplay.name}</h1>
+      <button class="favorite-btn" id=${recipeToDisplay.id}>Favorite Recipe</button>
+      <h2>Cost: ${recipeCost}</h2>
+      <h3>Ingredients: </h3>
+      <p>${officialIngredients}</p>
+      <h3>Instructions: </h3>
+      <p>${officialInstructions}</p>
+      `
+    }
+  })
+}
+
+function displayFavoriteRecipeArea() {
+  foodImagesSection.innerHTML = ''
+  foodImagesSection.classList.add("hidden");
+  allRecipesSection.classList.add("hidden");
+  viewAllButton.classList.add("hidden");
+  displayRecipeSection.classList.add("hidden");
+  displayRecipeSection.innerHTML = '';
+  filteredByTagArea.innerHTML= "";
+  favoriteRecipeArea.classList.remove("hidden")
+}
+
```

Comment:
Nice use of helper functions crew. We'll see if we can use more on final refactor.